File: leetcode/88.合并两个有序数组.go
/*
 * @Author: Opynicus
 * @Date: 2022-01-11 13:42:03
 * @LastEditTime: 2022-01-11 14:05:12
 * @LastEditors: Opynicus
 * @Description: 
 * @FilePath: \leetcode\88.合并两个有序数组.go
 * 可以输入预定的版权声明、个性签名、空行等
 */
/*
 * @lc app=leetcode.cn id=88 lang=golang
 *
 * [88] 合并两个有序数组
 */

// @lc code=start
func merge(nums1 []int, m int, nums2 []int, n int)  {
	nums3 := make([]int, 0, m+n)
	i, j := 0, 0
	for  true {
		if i == m {
			nums3 = append(nums3, nums2[j:]...)
			break
		}
		if j == n {
			nums3 = append(nums3, nums1[i:]...)
			break
		}
		if nums1[i] < nums2[j] {
			nums3 = append(nums3, nums1[i])
			i++
		}
		if nums1[i] >= nums2[j] {
			nums3 = append(nums3, nums2[j])
			j++
		}
	}
	copy(nums1, nums3)
}
// @lc code=end

File: leetcode/88.合并两个有序数组.cpp
/*
 * @Author: Opynicus
 * @Date: 2022-04-07 20:01:27
 * @LastEditTime: 2022-04-07 20:13:29
 * @LastEditors: Opynicus
 * @Description: 
 * @FilePath: \undefinede:\JuniorWork\leetcode\88.合并两个有序数组.cpp
 * 可以输入预定的版权声明、个性签名、空行等
 */
/*
 * @lc app=leetcode.cn id=88 lang=cpp
 *
 * [88] 合并两个有序数组
 */

// @lc code=start
class Solution {
public:
    void merge(vector<int>& nums1, int m, vector<int>& nums2, int n) {
        int i, j, k;
        vector<int> tmp(m+n);
        for (i = 0, j = 0, k = 0; i < m && j < n; k++)
        {
            if (nums1[i] <= nums2[j])
            {
                tmp[k] = nums1[i];
                i++;
            }
            else
            {
                tmp[k] = nums2[j];
                j++;
            }
        }
        while (i < m)
        {
            tmp[k] = nums1[i];
            k++;
            i++;
        }
        while (j < n)
        {
            tmp[k] = nums2[j];
            k++;
            j++;
        }
        for (int i = 0; i < m + n; i++)
        {
            nums1[i] = tmp[i];
        }
        
    }
};
// @lc code=end

